### ISSUE TITLE
section on `use_query` outdated?

### ISSUE THREAD

**Issue by ydirson** [leptos-rs/book#61]
Basing my tests on [the book](https://book.leptos.dev/router/18_params_and_queries.html) I wrote the following
```
use leptos::*;
use leptos_router::*;

#[derive(Params)]
struct UrlQuery {
    armies: String,
}

#[component]
fn App() -> impl IntoView {
    let query = use_query::<UrlQuery>();
```

First error is I get is `UrlQuery` should implement `PartialEq` for use with `use_query` - the book example does not use it, is the doc just a bit outdated, or is there anything more subtle than that?

And then
```
39 |     armies: String,
   |             ^^^^^^ the trait `IntoParam` is not implemented for `std::string::String`
```

In https://docs.rs/leptos_router/latest/leptos_router/trait.IntoParam.html we see it is only implemented for `Option<T>` 

Indeed using `Option<String>` allows the code to compile, so I guess the example should really be updated.

**Comment by gbj** [comment]
Thanks for your note. The implementations for non-`Option<_>` types are limited to `nightly` at present, and the book assumes you are using the `nightly` feature. It is worth adding a note here to clarify, and making sure that the docs.rs docs note the availability of `IntoParam` for additional types when using the `nightly` feature.

**Comment by ydirson** [comment]
Oh I see. With Leptonic not supporting `nightly` yet, it is sometimes hard for the new Leptos user to have the doc so centered on `nightly` :)

**Comment by gbj** [comment]
Closed as of 0.7 book update.
